config/crd/crd.projectcalico.org_blockaffinities.yaml

```diff
@@ -39,7 +39,7 @@ spec:
               cidr:
                 type: string
               deleted:
-                type: string
+                type: boolean
```

Comment:
This change does, unfortunately, mean that older clients (e.g., Calico v3.14 and older) won't be able to write the CRD any longer, because they don't have "omitempty" set so will try to write a string into a boolean field.

The alternative is that we keep it a string, and write some conversion between string / bool in the new code.

Reply:
What happens if there's data in the API server that is now no longer valid?   Are we allowed to read it back?

I'm inclined to keep it as a string with a comment to say the type is a historic mistake; it is an internal datastructure after all.

Reply:
> What happens if there's data in the API server that is now no longer valid? Are we allowed to read it back?

yeah I tested this exact thing yesterday. We're allowed to read it back fine. We just can't write it.